tests/drivers/mtd_partition/native_posix.overlay

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2021 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/delete-node/ &boot_partition;
+/delete-node/ &slot0_partition;
+
+&flash0 {
+        partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+                boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x000000000 0x0000C000>;
+			partitions {
+				compatible = "fixed-partitions";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				bootsub_partition: partition@4000 {
```

Comment:
oh, a sub-partition. Why it is needed?

Reply:
On linux there is no limitation that there can only be partitions and not subpartitions. Introducing subpartitions (this has always been possible, but there was no support to use them) will make a lot of things easier.

e.g. For image generation: It is possible to specify image0 as header + real_image0, generate the image for real_image0, upload to slot image0. When other bootloaders use a different scheme -> no problem. Very versatile.